notebook/base/handlers.py

```diff
@@ -427,11 +427,15 @@ def check_host(self):
         if host.startswith('[') and host.endswith(']'):
             host = host[1:-1]
 
+        if not PY3:
+            # ip_address only accepts unicode on Python 2
+            host = host.decode('utf8', 'replace')
+
         try:
             addr = ipaddress.ip_address(host)
         except ValueError:
             # Not an IP address: check against hostnames
-            allow = host in self.settings.get('local_hostnames', [])
+            allow = host in self.settings.get('local_hostnames', ['localhost'])
```

Comment:
additionally, added 'localhost' as the default here so that it is defined by default when used outside NotebookApp, e.g. nbdime.